src/Simulation/Simulators/QuantumSimulator/Measure.cs

```diff
@@ -31,10 +31,11 @@ public QSimMeasure(QuantumSimulator m) : base(m)
                 {
                     throw new InvalidOperationException($"Both input arrays for {this.GetType().Name} (paulis,qubits), must be of same size");
                 }
-                foreach (Qubit q in qubits)
+                if (qubits.Length == 1)
```

Comment:
Does it matter what basis the qubit is measured in? Do you want to release after _any_ single qubit measurement?

Reply:
Yeah, I this @cgranade put this best in the discussion for #231:

>> Releasing a Qubit measured in a different base, the qubit is still in super-position on the Z bases but it doesn't throw:
>>    ```
>>      using (q = Qubit()) {
>>          H(q);
>>          let _ = Measure([PauliX], [q]);
>>      }
>> ```
> Quick comment, but this should be entirely OK, as the qubit is still in a deterministically known state at the point that it is released, such that appropriate instructions can be automatically added to reset it to the |0⟩ state.

Reply:
This has been answered is @swernli 's post below.